src/less/tooltip/ds6/tooltip.less

```diff
@@ -0,0 +1,404 @@
+
+        &::after {
+            border-style: solid;
+            border-width: 8px;
+            content: "";
+            position: absolute;
+        }
+    }
+
+    &__body {
+        margin: 16px 16px 16px 16px;
+    }
+}
+
+.tooltip {
+    &:hover {
+        .tooltip__top-arrow,
+        .tooltip__top-leftarrow,
+        .tooltip__top-rightarrow,
+        .tooltip__bottom-arrow,
+        .tooltip__bottom-leftarrow,
+        .tooltip__bottom-rightarrow,
+        .tooltip__left-arrow,
+        .tooltip__left-toparrow,
+        .tooltip__left-bottomarrow,
+        .tooltip__right-arrow,
+        .tooltip__right-toparrow,
+        .tooltip__right-bottomarrow {
+            visibility: visible;
+        }
+    }
+
+    &__bottom-arrow,
+    &__bottom-leftarrow,
+    &__bottom-rightarrow,
+    &__top-arrow,
+    &__top-leftarrow,
+    &__top-rightarrow,
+    &__left-arrow,
+    &__left-toparrow,
+    &__left-bottomarrow,
+    &__right-arrow,
+    &__right-toparrow,
+    &__right-bottomarrow {
+        width: 344px;
+
+        &::after {
+            border-color: @ds6-color-p002-blue;
+            box-shadow: 0 0 0 @ds6-color-p002-blue;
+            box-sizing: border-box;
+        }
+    }
+
+    &__bottom-arrow,
+    &__bottom-leftarrow,
+    &__bottom-rightarrow {
+        bottom: 100%;
+        margin-bottom: 16px;
+
+        &::after {
+            top: 100%;
+        }
+    }
+
+    &__top-arrow,
+    &__top-leftarrow,
+    &__top-rightarrow {
+        margin-top: 16px;
+
+        &::after {
+            top: 2%;
+        }
+    }
+
+    &__left-arrow,
+    &__left-toparrow,
+    &__left-bottomarrow {
+        left: 110%;
+        margin-left: 16px;
+
+        &::after {
+            margin-left: -10px;
+        }
+    }
+
+    &__right-arrow,
+    &__right-toparrow,
+    &__right-bottomarrow {
+        margin-right: 16px;
+        right: 100%;
+
+        &::after {
+            left: 95%;
+            margin-left: 5px;
+        }
+    }
+
+    &__bottom-arrow,
+    &__top-arrow {
+        margin-left: -161px;
+
+        &::after {
+            left: 45%;
+            margin-left: 5px;
+        }
+    }
+
+    &__bottom-leftarrow,
+    &__top-leftarrow {
+        margin-left: -24px;
+
+        &::after {
+            left: 7%;
+        }
+    }
+
+    &__bottom-rightarrow,
+    &__top-rightarrow {
+        margin-left: -300px;
+
+        &::after {
+            left: 87%;
+        }
+    }
+
+    &__left-arrow,
+    &__right-arrow {
+        top: -100%;
+
+        &::after {
+            top: 50%;
+        }
+    }
+
+    &__left-toparrow,
+    &__right-toparrow {
+        top: -50%;
+
+        &::after {
+            top: 35%;
+        }
+    }
+
+    &__left-bottomarrow,
+    &__right-bottomarrow {
+        top: -140%;
+
+        &::after {
+            top: 65%;
+        }
+    }
+
+    &__window {
+        background-color: @ds6-color-p002-blue;
+        color: @ds6-color-g201-gray;
+        visibility: hidden;
+        width: 344px;
```

Comment:
Please use `max-width`.

Reply:
I talked to @prabhurangan about this.  The problem with making it max-width rather than width is that it doesn't automatically flex to the max width since its absolutely positioned.  It just get's taller. There may be some workaround, but I haven't played with it.

![screen shot 2018-09-06 at 3 06 13 pm](https://user-images.githubusercontent.com/1562843/45187746-a4c88800-b1e6-11e8-9cf5-0c2ff9c9112e.png)

Reply:
Oh. Maybe:

```
max-width: 344px;
width: 100%;
```

Reply:
Tried that.  

But @prabhurangan  it looks like we may be able to change the structure a little.  If we add another container around it and that container is has a max-width of 344px.  Then the internal container doesn't have a max-width or width set.  It looks like that might work.

![screen shot 2018-09-06 at 3 17 38 pm](https://user-images.githubusercontent.com/1562843/45188127-04736300-b1e8-11e8-8c94-5d3bdbe04e22.png)

![screen shot 2018-09-06 at 3 18 02 pm](https://user-images.githubusercontent.com/1562843/45188157-1bb25080-b1e8-11e8-9110-0cd69069d3e7.png)

So:
```
<div style="position:absolute; max-width: 344px">
    <div style="position:absolute">
        <p>Content</p>
    </div>
</div>
```